File: Java/refactoring_exercise_3/src/test/java/za/co/entelect/refactoring3/service/ImageServiceTest.java
package za.co.entelect.refactoring3.service;

import org.junit.Before;
import org.junit.Test;
import za.co.entelect.refactoring3.domain.Image;

import static junit.framework.Assert.*;

public class ImageServiceTest {

    public static final String IMAGE_1 = "image_1";
    public static final String IMAGE_2 = "image_2";

    public static final byte[] DATA = new byte[1];
    private ImageService imageService = new ImageService();

    @Before
    public void init(){
        imageService = new ImageService();
    }

    @Test
    public void testAddImage(){
        assertEquals(1, imageService.count());
        imageService.add(new Image(IMAGE_1, DATA));
        imageService.add(new Image(IMAGE_2, DATA));
        assertEquals(3, imageService.count());
    }

    @Test
    public void testFetchImage(){
        assertNull(imageService.fetch(IMAGE_1));
        imageService.add(new Image(IMAGE_1, new byte[1]));
        Image image1 = imageService.fetch(IMAGE_1);
        assertNotNull(image1);
        assertEquals(IMAGE_1, image1.getImageId());
        assertNotNull(image1.getData());
        assertNotNull(imageService.fetch(ImageService.DEFAULT_IMAGE));
    }
}

File: C#/Refactoring/refactoring_exercise_final/test/za/co/entelect/refactoring_final/service/ImageServiceTest.cs
using NUnit.Framework;
using NUnit.Framework.Constraints;
using refactoring_exercise_final.za.co.entelect.refactoring_final.domain;
using refactoring_exercise_final.za.co.entelect.refactoring_final.service;

namespace refactoring_exercise_final.test.za.co.entelect.refactoring_final.service
{
    [TestFixture]
    public class ImageServiceTest {

        public const string Image1 = "image_1";
        public const string Image2 = "image_2";

        public static readonly byte[] Data = new byte[1];
        private ImageService _imageService = new ImageService();

        [SetUp]
        public void Init(){
            _imageService = new ImageService();
        }

        [Test]
        public void TestAddImage(){
            Assert.That(_imageService.Count(), new EqualConstraint(1));
            _imageService.Add(new Image(Image1, Data));
            _imageService.Add(new Image(Image2, Data));
            Assert.That(_imageService.Count(), new EqualConstraint(3));
        }

        [Test]
        public void TestFetchImage(){
            Assert.Null(_imageService.Fetch(Image1));
            _imageService.Add(new Image(Image1, new byte[1]));
            Image image1 = _imageService.Fetch(Image1);
            Assert.NotNull(image1);
            Assert.That(image1.ImageId, new EqualConstraint(Image1));
            Assert.NotNull(image1.Data);
            Assert.NotNull(_imageService.Fetch(ImageService.DefaultImage));
        }
    }
}